### ISSUE TITLE
HLog overflow when varlen records is updated inplace in memory

### ISSUE THREAD

**Issue by marius-klimantavicius** [microsoft/FASTER#167]
Steps to reproduce (variable length structs):
1. Upsert key1:value1
1. Upsert key2:value2 (it should create a record just after first key1:value1)
1. Upsert key1:value3 where `length(value3) >length(value1)` (must be substantially longer than original value1 and average; the records in question must still be within mutable memory region).

Expected result:
A new record is created because there is not enough space in the first record.

Actual result:
Record is updated in place thus overwrites record of key2:value2 (buffer overflow of sorts).

**Comment by marius-klimantavicius** [comment]
PR #164 contains a possible fix for this issue.

**Comment by badrishc** [comment]
Yeah, this is a known limitation in the C# version. The C++ version actually supports this. Read this for how it is handled in C++: https://github.com/Microsoft/FASTER/wiki/Variable-length-values
